### ISSUE TITLE
ASP.NET Core 1.0.1 on IIS fails with HTTP Error 502.5 - Process Failure

### ISSUE THREAD

**Issue by b0bi79** [aspnet/IISIntegration#283]
Hello.
I get HTTP Error 502.5 - Process Failure. After added disableStartUpErrorPage="true" stdoutLogEnabled="true" in the web.config, I get HTTP Error 502.3 - Bad Gateway. 
Windows event log: Process was created with commandline '"C:\inetpub\wwwroot\VernoPortal\Verno.Reports.Web.exe" ' but either crashed or did not reponse within given time or did not listen on the given port '13417', ErrorCode = '0x800705b4'
ASP.NET Core Module Log creates an empty log file.
Verno.Reports.Web.exe started from console work fine.

Site deployed on Windows Server 2008 R2 with Web Deploy using Visual Studio. 
On the server are installed:
- DotNetCore.1.0.1-WindowsHosting
- httpPlatformHandler_amd64
- .NET 4.6.2

Web.Config
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath=".\Verno.Reports.Web.exe" arguments="" disableStartUpErrorPage="true" stdoutLogEnabled="true" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false" />
  </system.webServer>
</configuration>
```

project.json
```
{
  "dependencies": {
    "Microsoft.Owin": "3.0.1",
    "Microsoft.AspNetCore.Owin": "1.0.0",
    "Microsoft.AspNetCore.Authentication.Cookies": "1.0.0",
    "Microsoft.AspNetCore.Diagnostics": "1.0.0",
    "Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore": "1.0.0",
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
    "Microsoft.AspNetCore.StaticFiles": "1.0.0",
    "Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
    },
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0",
    "Microsoft.Extensions.Configuration.Json": "1.0.0",
    "Microsoft.Extensions.Logging": "1.0.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0",
    "System.Linq": "4.1.0",
    "Microsoft.AspNetCore.Mvc": "1.0.1",
    "Microsoft.EntityFrameworkCore.SqlServer": "1.0.1",
    "Microsoft.EntityFrameworkCore.SqlServer.Design": "1.0.1",
    "Microsoft.VisualStudio.Web.CodeGeneration.Tools": "1.0.0-preview2-update1",
    "Microsoft.VisualStudio.Web.CodeGenerators.Mvc": "1.0.0-preview2-update1"
  },

  "tools": {
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final",
    "Microsoft.EntityFrameworkCore.Tools": "1.0.0-preview2-final"
  },

  "frameworks": {
    "net461": {}
  },

  "buildOptions": {
    "emitEntryPoint": true,
    "preserveCompilationContext": true,
    "copyToOutput": "log4net.config"
  },

  "publishOptions": {
    "include": [
      "wwwroot",
      "Views",
      "appsettings.json",
      "web.config",
      "log4net.config"
    ]
  },

  "scripts": {
    "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
  }
}
```
Help me please.

**Comment by Tratcher** [comment]
https://docs.asp.net/en/latest/publishing/iis.html#common-errors
https://docs.asp.net/en/latest/hosting/aspnet-core-module.html#asp-net-core-module-configuration-examples

**Comment by b0bi79** [comment]
@Tratcher Thanks Chris. By common-errors guide I shoud call the .UseIISIntegration and .UseUrls() extension method positioned before the .UseIISIntegration(). Here my middleware:
```
                var host = new WebHostBuilder()
                    .UseKestrel()
                    .UseContentRoot(Directory.GetCurrentDirectory())
                    .UseIISIntegration()
                    .UseStartup<Startup>()
                    .Build();
```
And I have issue only on Windows Server 2008 R2. On my develop computer with Win7 IIS works fine.

**Comment by arthis** [comment]
Same thing here. Running dotnet  myLibrary.dll works fine

**Comment by GuardRex** [comment]
@b0bi79 Since the process is created but the app doesn't respond, the issue will probably be within the app startup/configuration. You can confirm your IIS/ANCM is configured and working well with a test app (e.g., https://github.com/aspnet/cli-samples/tree/master/HelloMvc).

Side Note: The HTTP Platform Handler is depreciated. It won't hurt anything (probably) to leave it on there, but HPH is not required, not recommended, and not supported.

**Comment by b0bi79** [comment]
I solved this issue by removing DotNetCore, httpPlatformHandler_amd64, .NET 4.6.2 and reinstalled this. But instead .NET 4.6.2 I installed .NET 4.6.1

**Comment by arthis** [comment]
Pb solced, I had to restart iis after installation of asp.net core module.



On Thu, Oct 20, 2016 at 10:51 AM +0200, "Rozhkov Alexey" <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #283<https://github.com/aspnet/IISIntegration/issues/283>.

-
